annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+                        {"id": "5", "status": "Complete"},
+                    ]
+                },
+                4: {
+                    "objects": [
+                        {"id": "6", "description": "Task 6"},
+                    ]
+                },
+            },
+            {
+                3: [({"name": "Task 4"}, "4"), ({"status": "Complete"}, "5")],
+                4: [({"description": "Task 6"}, "6")],
+            },
+        ),
+    ),
+)
+def test_get_tasks_without_ids(
+    tasks: Dict[int, Any], expected: Dict[int, Any]
+):
+    assert get_tasks_without_ids(tasks) == expected
+
+
+@pytest.fixture
+def mock_dependencies():
+    with patch(
+        "annotation.tasks.services.get_file_path_and_bucket",
+        return_value=("s3/path", "bucket"),
+    ) as mock_get_file_path_and_bucket, patch(
+        "annotation.tasks.services.get_annotation_tasks", return_value={}
+    ) as mock_get_annotation_tasks, patch(
+        "annotation.tasks.services.construct_annotated_pages",
+        return_value=([], set()),
+    ) as mock_construct_annotated_pages, patch(
+        "annotation.tasks.services.construct_annotated_doc", return_value=None
+    ) as mock_construct_annotated_doc, patch(
+        "annotation.tasks.services.update_task_status", return_value=None
+    ) as mock_update_task_status, patch(
+        "annotation.tasks.services.Logger.exception", return_value=None
+    ) as mock_logger_exception:
+
+        yield {
+            "mock_get_file_path_and_bucket": mock_get_file_path_and_bucket,
+            "mock_get_annotation_tasks": mock_get_annotation_tasks,
+            "mock_construct_annotated_pages": mock_construct_annotated_pages,
+            "mock_construct_annotated_doc": mock_construct_annotated_doc,
+            "mock_update_task_status": mock_update_task_status,
+            "mock_logger_exception": mock_logger_exception,
+        }
+
+
+def test_create_validation_revisions(mock_dependencies: Dict[str, Mock]):
+    mock_db = MagicMock()
+    x_current_tenant = "tenant_1"
+    token = MagicMock(token="fake_token")
+    job_id = 123
+    validation_tasks = [
+        ManualAnnotationTask(file_id=1, pages={1}, user_id=1, id=10)
+    ]
+
+    mock_dependencies["mock_construct_annotated_pages"].return_value = (
+        [MagicMock(objs=True)],
+        {"category1"},
+    )
+    mock_dependencies["mock_construct_annotated_doc"].return_value = None
+    mock_dependencies["mock_update_task_status"].return_value = None
+
+    with patch(
+        "annotation.tasks.services.DocForSaveSchema"
+    ) as mock_doc_for_save_schema:
+        mock_doc_instance = mock_doc_for_save_schema.return_value
+        mock_doc_instance.user = validation_tasks[0].user_id
+        mock_doc_instance.pages = [MagicMock(objs=True)]
+        mock_doc_instance.validated = set()
+        mock_doc_instance.failed_validation_pages = set()
+        mock_doc_instance.categories = {"category1"}
```

Comment:
It seems to me that all of this was done to work around the lack of user_id in validation_tasks, which you could have passed in while instantiating.